jetty-server/src/main/java/org/eclipse/jetty/server/SymlinkAllowedResourceAliasChecker.java

```diff
@@ -0,0 +1,72 @@
+//
+//  ========================================================================
+//  Copyright (c) 1995-2021 Mort Bay Consulting Pty Ltd and others.
+//  ------------------------------------------------------------------------
+//  All rights reserved. This program and the accompanying materials
+//  are made available under the terms of the Eclipse Public License v1.0
+//  and Apache License v2.0 which accompanies this distribution.
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
+//
+//      The Apache License v2.0 is available at
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+package org.eclipse.jetty.server;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import org.eclipse.jetty.server.handler.ContextHandler;
+import org.eclipse.jetty.util.log.Log;
+import org.eclipse.jetty.util.log.Logger;
+import org.eclipse.jetty.util.resource.Resource;
+
+/**
+ * An extension of {@link AllowedResourceAliasChecker} which only allows resources if they are symlinks.
+ */
+public class SymlinkAllowedResourceAliasChecker extends AllowedResourceAliasChecker
+{
+    private static final Logger LOG = Log.getLogger(SymlinkAllowedResourceAliasChecker.class);
+
+    /**
+     * @param contextHandler the context handler to use.
+     */
+    public SymlinkAllowedResourceAliasChecker(ContextHandler contextHandler)
+    {
+        super(contextHandler, false);
+    }
+
+    @Override
+    public boolean check(String uri, Resource resource)
+    {
+        try
+        {
+            // Check the resource is allowed to be accessed.
+            if (!super.check(uri, resource))
+                return false;
+
+            // Only approve resource if it is accessed by a symbolic link.
+            Path resourcePath = resource.getFile().toPath();
+            if (Files.isSymbolicLink(resourcePath))
+                return true;
+
+            // TODO: If base resource contains symlink then this will always return true.
+            //  But we don't want to deny all paths if the resource base is symbolically linked.
+            if (super.hasSymbolicLink(resourcePath))
```

Comment:
Call a varient of this method that only checks up to the resourceBase for a symbolicLink

Reply:
That would cause the issue I described in the comment. If we do this and the base resource contains a symlink we will deny all files under the base resource, even though the only reason they are an alias is because the base resource contains symlink.

So doing that would break the capability of symlinking the base resource directory.

Reply:
The purpose of this checker is to allow aliases if they are symlinks within to scope of the webapp.   
So in the case of `/var/lib/jetty-home/webapps/mycontext/foo/bar/bob.txt` we would walk up the parent tree and if any of the following:
 - `/var/lib/jetty-home/webapps/mycontext/foo/bar/bob.txt`
 - `/var/lib/jetty-home/webapps/mycontext/foo/bar`
 - `/var/lib/jetty-home/webapps/mycontext/foo`

Are symlinks, then we allow the alias.   If the alias is above that point, then we don't even look.  It is not allowed.

Reply:
I am fine with this change, but it is worth noting that it is a possible breaking change for people who are already relying of that behaviour from `AllowSymLinkAliasChecker`.

So if they run into this issue they can be directed to configure `AllowedResourceAliasChecker` which will allow this use case.